lib/osf-components/addon/components/file-browser/folder-item/template.hbs

```diff
@@ -1,49 +1,68 @@
+    <li
+        data-test-file-list-item={{@item.id}}
+        data-test-indented={{if @manager.parentFolder 'true' 'false'}}
+        local-class='FileList__item {{if @manager.parentFolder 'Indent'}} {{if isSelected 'Selected'}}'
     >
-        <FaIcon
-            @icon='folder' @fixedWidth={{true}}
-        />
-        <span data-test-folder-name>
-            {{@item.name}}
-        </span>
-    </Button>
-    <div
-        data-test-file-list-date
-        local-class='FileList__item__options NoShrink'
-    >
-        <ResponsiveDropdown
-            local-class='DownloadShareDropdown'
-            @renderInPlace={{true}}
-            @buttonStyling={{true}}
-            @horizontalPosition='right'
-            as |dropdown|
+        {{#if @selectable}}
+            <label
+                data-analytics-name='Select folder'
+                data-analytics-category='checkbox'
+                data-test-select-folder={{@item.id}}
+                aria-label={{t 'osf-components.file-browser.select_folder' folderName=@item.name}}
+                local-class='Label {{if isSelected 'Selected'}}'
+            >
+                <Input
+                    local-class='Checkbox'
+                    @type='checkbox'
+                    @checked={{isSelected}}
+                    {{on 'click' (action @manager.selectFile @item)}}
+                />
+            </label>
+        {{/if}}
+
+        <Button
+            data-test-file-list-link
+            data-analytics-name='View folder'
+            @layout='fake-link'
+            local-class='FileList__item__options FolderButton'
+            {{on 'click' (perform @manager.goToFolder @item)}}
+        >
+            <FaIcon
+                @icon='folder' @fixedWidth={{true}}
+            />
+            <span data-test-folder-name>
+                {{@item.name}}
+            </span>
+        </Button>
+        <div
+            data-test-file-list-date
+            local-class='FileList__item__options NoShrink'
         >
-            <dropdown.trigger
-                aria-label={{t 'file_actions_menu.actions' filename=@item.name}}
-                data-test-file-download-share-trigger
-                data-analytics-name='Folder actions'
+            <ResponsiveDropdown
+                local-class='DownloadShareDropdown'
+                @renderInPlace={{true}}
+                @buttonStyling={{true}}
+                @horizontalPosition='right'
+                as |dropdown|
             >
-                <FaIcon @icon='ellipsis-v' />
-            </dropdown.trigger>
-            <dropdown.content local-class='DropdownList'>
-                <OsfLink
-                    @href={{@item.links.download}}
-                    data-test-download-button
+                <dropdown.trigger
+                    aria-label={{t 'file_actions_menu.actions' filename=@item.name}}
+                    data-test-file-download-share-trigger
+                    data-analytics-name='Folder actions'
```

Comment:
This is a potential add as well, but if the dropdown is labeled it may be excessive. It still helps though.